File: mobile/app/src/main/java/com/example/colorimagemobile/model/Message.kt
package com.example.colorimagemobile.model

class Message(message: String, timestamp: String, author: String, roomName: String) {
    var message: String = ""
    var timestamp: String = ""
    var author: String = ""
    var roomName: String = ""

    init {
        this.message = message
        this.timestamp = timestamp
        this.author = author
        this.roomName = roomName
    }
}

File: server/app/domain/models/Message.ts
import mongoose, { Document, Model } from 'mongoose';

export interface MessageInterface extends Document{
    message: string, 
    timestamp: string, 
    author: string,
    roomName: string
}

const MessageSchema = new mongoose.Schema({
	message: { type: String, required: true },
    timestamp: { type: String, required: true },
	author: { type: String, required: true },
    roomName: { type: String, required: true, ref: 'TextChannel' },
});

export const Message: Model<MessageInterface> = mongoose.model('Messages', MessageSchema);